src/bullet_trace/BulletTrace.gd

```diff
@@ -0,0 +1,54 @@
+@icon("res://src/assets/noexport/artillery-shell.svg")
+class_name BulletTrace
+extends Line2D
+## Class for handling bullet trace animations, using the [Line2D] logic. [br]
+## An object of this class must always have a shader with a [code]lifetime[/code] and a [code]life[/code] variable.
+
+## Value dictating how far the bullet travels by default.
+## TODO: Make this variable weapon specific.
+static var ray_distance: float = 1000.0
+## Value dictating the scale of the random angle offset that occurs when a bullet is fired.
+## TODO: Make this variable weapon specific.
+static var ray_angle_offset: float = 0.04
+## Value dictating how long a bullet trace lingers before vanishing.
+## Should be scaled with the [code]ray_distance[/code] variable.
+static var trace_lifetime_factor: float = ray_distance * 2.0
```

Comment:
Why are these static?

Reply:
Cuz the function that uses them is also static. This might just be my brain on java, so if this isn't needed, tell me tyvm